portfolio/src/main/java/com/google/sps/servlets/CovidDataServlet.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that retrieves data from csv files*/
+@WebServlet("/get-covidData")
+public class CovidDataServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    LinkedHashMap<String, LinkedHashMap<String, Integer>> covidData = 
+        new LinkedHashMap<String, LinkedHashMap<String, Integer>>();
```

Comment:
Just curious why the outer map needs to be a LinkedHashMap as well? For the data it makes sense because you want to maintain iteration order, but it seems like the sets of data could be accessed in any order.

Also, in general we instantiate an instance of the interface, like:

```Java
Map<String, Integer> myMap = new HashMap<String, Integer>();
```

or

```Java
List<String> myList = new ArrayList<String>();
```

so for this case you might want something like:

```Java
Map<String, LinkedHashMap<String, Integer>> covidData = new LinkedHashMap<String, LinkedHashMap<String, Integer>>();
```

(or whatever you decide to go with)

Reply:
Hmmm, true. So instead of using a LinkedHashMap for the outer map, a HashMap would be okay, right? Since the order of insertion does not matter for the outer map.

And thanks for pointing out the general way to instantiate, I wasn't aware of it!

Reply:
Yes I think a HashMap would work fine since you only need (key,value) pairs, no linking between.